package.json

```diff
@@ -22,6 +22,15 @@
   },
   "license": "MIT",
   "author": "stylelint",
+  "type": "module",
```

Comment:
question

> Regardless of the value of the "type" field, .mjs files are always treated as ES modules and .cjs files are always treated as CommonJS.

Why do we need `type`?
It's only useful for `.js` files; I don't think we have any anymore.

Is it for other tools?
Or to declare our intent to transition to pure ESM?

Reply:
> to declare our intent to transition to pure ESM in the next major?

This is my intent, but `type` is actually unneeded if there are any downsides.

Reply:
> any downsides

If you are planning to add `.js` files in the future, remove it.
Adding `type` without any reason seems weird to me, but I am sure it will have consequences whether it's missing or not. I hope the next tag will help us resolve that.

minor

If you keep it, we will need to add somewhere in the documentation something about always using either `.cjs` or `.mjs` until v17; no need to do it in this PR though. Or alternatively just use [this rule](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/extensions.md) with `"js": "never"`.

Reply:
Make sense. Removed `type` with 9a6621c35.

Reply:
For the record, I am not sure either way.
Let's wait for someone to request it or v17.